File: internal/runtime/status.go
package runtime

type Phase string

const (
	PhaseIdle         Phase = "idle"
	PhaseListening    Phase = "listening"
	PhaseHandshaking  Phase = "handshaking"
	PhaseReady        Phase = "ready"
	PhaseDisconnected Phase = "disconnected"
	PhaseError        Phase = "error"
)

type Status struct {
	Target         string `json:"target"`
	Phase          Phase  `json:"phase"`
	Connected      bool   `json:"connected"`
	Ready          bool   `json:"ready"`
	InstanceID     string `json:"instanceId,omitempty"`
	HostVersion    string `json:"hostVersion,omitempty"`
	LastSeenAt     string `json:"lastSeenAt,omitempty"`
	ProgressDetail string `json:"progressDetail,omitempty"`
	LastError      string `json:"lastError,omitempty"`
}

func InitialStatus() Status {
	return Status{
		Target: string(RuntimeTargetQQWS),
		Phase:  PhaseIdle,
	}
}

File: frontend/src/lib/status.ts
export type PhaseTone = 'good' | 'warn' | 'bad' | 'neutral';

const labels: Record<string, string> = {
  idle: '空闲',
  listening: '正在监听',
  handshaking: '握手中',
  ready: '已就绪',
  disconnected: '已断开',
  error: '错误',
};

export function phaseLabel(phase: string) {
  return labels[phase] ?? phase;
}

export function phaseTone(phase: string): PhaseTone {
  if (phase === 'ready' || phase === 'listening') return 'good';
  if (phase === 'handshaking') return 'warn';
  if (phase === 'error' || phase === 'disconnected') return 'bad';
  return 'neutral';
}
